app/controllers/index.rb

```diff
@@ -0,0 +1,101 @@
+# displays home page with login prompt
+get '/' do
+  erb :index
+end
+
+# analyzes and stores login/registration info
+post '/' do
+  # checking if registration info is correct
+  if params[:new_password].blank? && params[:new_username].blank?
```

Comment:
These validations should be inside your model. Also your params should either be `params[:username]` `params[:password]` or  `params[:user][:username]` `params[:user][:password]`

```ruby
post '/' do
  @user = User.new(username: params[:username], password: params[:password], email: params[:email])
  if @user.save
    login_as_user(@user)
    redirect '/'
  else
    erb :create_user # use @user in the template to display errors from validations
  end
end
```